semgrep/semgrep/core_output.py

```diff
@@ -0,0 +1,325 @@
+from pathlib import Path
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import NewType
+from typing import Optional
+from typing import Tuple
+
+import attr
+
+from semgrep.error import Level
+from semgrep.error import SemgrepCoreError
+from semgrep.rule import Rule
+from semgrep.rule_match import CoreLocation
+from semgrep.rule_match import RuleMatch
+from semgrep.types import JsonObject
+from semgrep.verbose_logging import getLogger
+
```

Comment:
I have the same remark that I made to @nbrahms on his PR earlier. A short header placing this file in context would be appreciated (should answer "what's this code for?").